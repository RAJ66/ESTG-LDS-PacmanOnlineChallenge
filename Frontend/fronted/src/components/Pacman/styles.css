.pacman {
  width: 99px;
  height: 100px;
  border-radius: 50%;
  background: var(--color-second);
  position: relative;
  margin-top: var(--size-small-medium);
  margin-bottom: var(--size-small-medium);
  margin-left: var(--size-small-medium);
  margin-right: var(--size-small-medium);
}
.pacmanEye {
  position: absolute;
  width: var(--size-small);
  height: var(--size-small);
  border-radius: 50%;
  top: var(--size-small-medium);
  right: var(--size-medium);
  background: var(--color-first);
}

.pacmanMouth {
  background: var(--color-base);
  position: absolute;
  width: 100%;
  height: 100%;
  clip-path: polygon(100% 80%, 44% 48%, 100% 21%);

  animation-name: eat;
  animation-duration: 0.8s;
  animation-iteration-count: infinite;
}

@keyframes eat {
  0% {
    clip-path: polygon(100% 74%, 44% 48%, 100% 21%);
  }
  25% {
    clip-path: polygon(100% 60%, 44% 48%, 100% 40%);
  }
  50% {
    clip-path: polygon(100% 50%, 44% 48%, 100% 50%);
  }
  75% {
    clip-path: polygon(100% 59%, 44% 48%, 100% 35%);
  }
  100% {
    clip-path: polygon(100% 74%, 44% 48%, 100% 21%);
  }
}

.pacmanFood {
  position: absolute;
  width: 15px;
  height: 15px;
  background: var(--color-zero);
  border-radius: 50%;
  top: 40%;
  left: 120px;

  animation-name: food;
  animation-duration: 0.8s;
  animation-iteration-count: infinite;
}

@keyframes food {
  0% {
    transform: translateX(0);
    opacity: 1;
  }
  100% {
    transform: translateX(-50px);
    opacity: 0;
  }
}
