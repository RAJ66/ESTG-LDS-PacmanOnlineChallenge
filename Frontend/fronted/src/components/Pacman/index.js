import React from "react";

import "./styles.css";

export default function Pacman() {
  return (
    <div className="pacman">
      <div className="pacmanEye"></div>
      <div className="pacmanMouth"></div>
      <div className="pacmanFood"></div>
    </div>
  );
}
