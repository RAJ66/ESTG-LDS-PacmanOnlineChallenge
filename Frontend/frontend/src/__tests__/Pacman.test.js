import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Pacman from "../components/Pacman/index";

describe("Testing Pacman Component", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<Pacman />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
