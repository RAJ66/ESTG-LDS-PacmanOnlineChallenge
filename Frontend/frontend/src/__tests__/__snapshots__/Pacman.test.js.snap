// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pacman Component should render correctly 1`] = `
<div
  className="pacman"
>
  <div
    className="pacmanEye"
  />
  <div
    className="pacmanMouth"
  />
  <div
    className="pacmanFood"
  />
</div>
`;
